redis/tests/test_cluster.rs

```diff
@@ -747,3 +760,132 @@ fn test_cluster_split_multi_shard_command_and_combine_arrays_of_values() {
     let result = cmd.query::<Vec<String>>(&mut connection).unwrap();
     assert_eq!(result, vec!["foo-6382", "bar-6380", "baz-6380"]);
 }
+
+#[cfg(feature = "tls-rustls")]
+mod mtls_test {
+    use super::*;
+    use crate::support::mtls_test::create_cluster_client_from_cluster;
+    use redis::ConnectionInfo;
+
+    #[test]
+    fn test_cluster_basics_with_mtls() {
+        let cluster = TestClusterContext::new_with_mtls(3, 0);
+
+        let client = create_cluster_client_from_cluster(&cluster, true).unwrap();
+        let mut con = client.get_connection().unwrap();
+
+        redis::cmd("SET")
+            .arg("{x}key1")
+            .arg(b"foo")
+            .execute(&mut con);
+        redis::cmd("SET").arg(&["{x}key2", "bar"]).execute(&mut con);
+
+        assert_eq!(
+            redis::cmd("MGET")
+                .arg(&["{x}key1", "{x}key2"])
+                .query(&mut con),
+            Ok(("foo".to_string(), b"bar".to_vec()))
+        );
+    }
+
+    #[test]
+    fn test_cluster_should_not_connect_without_mtls() {
```

Comment:
What happens if you try to run a SET/GET using a new client that doesn't have TLS enabled? Can we make some assertions based on that?

Reply:
In this case it shouldn't even be able to create a connection, unless the server is only accepting TCP
Iiuc I believe that's already handled by this test case when credentials are not provided (`mtls_enabled:false`)

Reply:
Ok, yes, that makes sense -- thanks for clarifying!